### ISSUE TITLE
bug(dialog): NVDA does not announce “dialog” role when opening MatDialog with injected data

### ISSUE THREAD

**Issue by heyitstay11** [angular/components#33114]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

When opening a MatDialog with injected data, NVDA screen reader announces the dialog title and content, but does not announce that a dialog has opened (i.e., the word “dialog” is missing).

This issue is reproducible using the official Angular Material documentation example "Injecting data when opening a dialog".

### Reproduction

StackBlitz link: https://stackblitz.com/edit/vxadeyye?file=src%2Fexample%2Fdialog-data-example.ts 
(forked from Material Docs)

Steps to reproduce:
1. Enable NVDA screen reader.
2. Open the "Injecting data when opening a dialog" example:
3. Activate the `Open Dialog` button.
4. Listen to NVDA's announcement.

### Expected Behavior

When the dialog opens, NVDA should announce "Favorite Animal dialog"

### Actual Behavior

NVDA does not announce the word “dialog”

<img width="1877" height="897" alt="Image" src="https://github.com/user-attachments/assets/d7dc603f-249b-41df-bce1-ac097520d00a" />

### Environment

- Angular: 21.1.0
- CDK/Material: 21.1.0
- Browser(s): Chrome, Firefox
- Operating System (e.g. Windows, macOS, Ubuntu): Windows 11

**Comment by crisbeto** [comment]
I don't have access to NVDA to verify it, but I did notice the opposite problem: when the `data` is passed in we set `aria-labelledby` correctly, whereas when it's not, we fail to set it. Are you sure it's not the opposite issue where it only gets set with the `data` attribute?

**Comment by crisbeto** [comment]
Ah nevermind, I was seeing the difference because removing the `data` caused a runtime error in the template.

**Comment by heyitstay11** [comment]
When opening a dialog without an interactive element inside, NVDA does not announce the dialog role.
This is a known NVDA issue: https://github.com/nvaccess/nvda/issues/8620

No changes are required on "@angular/material" side. Closing this ticket accordingly.
